New pydocs, bugfix in base.html, rename field in model class and update migrations

DrEdi opened the pull request:
Add new expanded pydocs to some methods
Delete excessive string with bug in base.html

DrEdi committed:
Bugfix

Fix pydocs
Encapsulate filtering in Post main/models.py. Now it's in new method
Delete excessive string with bug in base.html

DrEdi committed:
Bugfix in Proficil

webknjaz commented:
The title is insane. Please fix it.

webknjaz commented on main/models.py:
```diff
@@ -25,8 +25,8 @@ def __str__(self):
     def approved_comments(self):
-        """Approving comments. Use for moderating."""
-        return self.comments.filter(approved_comment=True)
+        """Will show comments that are ok in your opinion."""
```
`Will show` is a Future Tense. You need Present Simple `Show`.
Not `your opinion`, but `someone's opinion`.

webknjaz commented on main/models.py:
```diff
@@ -36,13 +36,17 @@ class Comment(models.Model):
         """Render Comment instance as its text by default when stringifying."""
         return self.text
+
+    def get_approved(self):
```
I would call it `approved` and make it `@property`

webknjaz commented on main/models.py:
```diff
@@ -36,13 +36,17 @@ class Comment(models.Model):
         return self.text
+
+    def get_approved(self):
+        """Return comment if it is approved by the host."""
```
Please DON'T write docstrings explaining a `return` statement. Write HIGH-LEVEL description about what the method really DOES.

webknjaz commented on main/models.py:
```diff
@@ -36,13 +36,17 @@ class Comment(models.Model):
+
+    def get_approved(self):
+        """Return comment if it is approved by the host."""
+        return self if self.is_approved is True else None
```
This is insane. You should return a querystring (like in a previous version).

webknjaz commented on main/models.py:
```diff
@@ -25,8 +25,8 @@ def __str__(self):
-        """Approving comments. Use for moderating."""
-        return self.comments.filter(approved_comment=True)
+        """Will show comments that are ok in your opinion."""
+        return self.comments.get_approved()
```
Okay, it seems I misunderstood the context and the filtering needs to stay here. Please revert related changes.

webknjaz replied:
So I've googled a little, and recalled that this could be done using custom db manager. See the following for more info:
- https://docs.djangoproject.com/en/1.10/topics/db/managers/#modifying-a-manager-s-initial-queryset
- https://docs.djangoproject.com/en/1.10/topics/db/managers/#using-managers-for-related-object-access

webknjaz commented on main/models.py:
```diff
@@ -25,8 +25,8 @@ def __str__(self):
         return self.title
 
     def approved_comments(self):
```
I'd make it `@property`

webknjaz commented:
@DrEdi what is the relation between current title and changes done in this branch?

DrEdi committed:
Return filtering, fix pydoc

Returned filtering to models.py like in prev version
because we do not use class based views right now
so I think that could be better to change it later
when we'll upgrade blog to class-based views

webknjaz commented on main/templates/main/base.html:
```diff
@@ -14,7 +14,6 @@
         <a href="{% url 'post_new' %}" class="top-menu"><span class="glyphicon glyphicon-plus"></span></a>
         <a href="{% url 'post_draft_list' %}" class="top-menu"><span class="glyphicon glyphicon-edit"></span></a>
         <p class="top-menu">Hello {{ user.username }}<small>(<a href="{% url 'logout' %}">Log out</a>)</small></p>
-        <a href="{% url 'post_detail' pk=post.pk %}">Comments: {{ post.approved_comments.count }}</a>
```
I think you don't need to remove this line

DrEdi replied:
But there is a problem with this line:
`Reverse for 'post_detail' with arguments '()' and keyword arguments '{'pk': ''}' not found. 1 pattern(s) tried: ['post/(?P<pk>\\d+)/$']` if I try to see page logged. 
And this really have sense - we do not give any info to this template. Info about post's come only to post_detail, index etc.

webknjaz replied:
ok

webknjaz reviewed (changes requested):
Still waiting for you to:
1) cover all the comments
2) re-title and describe the PR

DrEdi committed:
Add @property to approved_comments

webknjaz commented on main/models.py:
```diff
@@ -36,11 +37,11 @@ class Comment(models.Model):
     text = models.TextField()
     created_date = models.DateTimeField(default=timezone.now)
-    approved_comment = models.BooleanField(default=False)
+    is_approved = models.BooleanField(default=False)
```
How does this relate to the PR title?

webknjaz commented on main/migrations/0003_auto_20161223_1550.py:
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
```
How does this relate to the PR title?

DrEdi replied:
retitled :)
